activerecord/test/cases/calculations_test.rb

```diff
@@ -681,4 +681,17 @@ def test_sum_uses_enumerable_version_when_block_is_given
     end
     assert block_called
   end
+
+  def test_having_with_strong_parameters
+    params = ActionController::Parameters.new(credit_limit: '50')
```

Comment:
We can't use ActionController here. Let just use the `ProtectedParams` but change it to not inherit from HWIA